service/policy/obligations/obligations.go

```diff
@@ -0,0 +1,158 @@
+package obligations
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"connectrpc.com/connect"
+	"github.com/opentdf/platform/protocol/go/policy/obligations"
+	"github.com/opentdf/platform/protocol/go/policy/obligations/obligationsconnect"
+	"github.com/opentdf/platform/service/logger"
+	"github.com/opentdf/platform/service/pkg/config"
+	"github.com/opentdf/platform/service/pkg/serviceregistry"
+	policyconfig "github.com/opentdf/platform/service/policy/config"
+	policydb "github.com/opentdf/platform/service/policy/db"
+)
+
+type Service struct {
+	dbClient policydb.PolicyDBClient
+	logger   *logger.Logger
+	config   *policyconfig.Config
+}
+
+func OnConfigUpdate(s *Service) serviceregistry.OnConfigUpdateHook {
+	return func(_ context.Context, cfg config.ServiceConfig) error {
+		sharedCfg, err := policyconfig.GetSharedPolicyConfig(cfg)
+		if err != nil {
+			return fmt.Errorf("failed to get shared policy config: %w", err)
+		}
+		s.config = sharedCfg
+		s.dbClient = policydb.NewClient(s.dbClient.Client, s.logger, int32(sharedCfg.ListRequestLimitMax), int32(sharedCfg.ListRequestLimitDefault))
+
+		s.logger.Info("obligations service config reloaded")
```

Comment:
Nit: you could use `s.logger.InfoContext(ctx, "obligations service config reloaded")` to use the `context.Context` argument and propagate it into the logs.

Reply:
updated